packages/clerk-js/src/utils/__tests__/url.test.ts

```diff
@@ -257,6 +256,49 @@ describe('buildURL(options: URLParams, skipOrigin)', () => {
       ),
     ).toBe('http://test.host/foo?my-search=42#my-hash/qux?my-hash-search-1=42&my-hash-search-2=42');
   });
+
+  it('appends search params passed to hashSearchParams in the URL fragment', () => {
+    const base = 'https://clerk.com/';
+    const params = new URLSearchParams({ test1: '1', test2: '2' });
+    const url = buildURL({ base, hashSearchParams: params }, { stringify: true });
+    expect(url).toBe('https://clerk.com/#/?test1=1&test2=2');
+  });
+
+  it('does not append a URL fragment if nothing was passed', () => {
+    const base = 'https://clerk.com/';
+    const url = buildURL({ base }, { stringify: true });
+    expect(url).toBe('https://clerk.com/');
+  });
+
+  it('does not append a URL fragment if search params were passed but were empty', () => {
+    const base = 'https://clerk.com/';
+    const params = new URLSearchParams({});
+    const url = buildURL({ base, hashSearchParams: params }, { stringify: true });
+    expect(url).toBe('https://clerk.com/');
+  });
+
+  it('appends search params to the fragment if search params is a plain object', () => {
+    const base = 'https://clerk.com';
+    const params = { test1: '1', test2: '2' };
+    const url = buildURL({ base, hashSearchParams: params }, { stringify: true });
+    expect(url).toBe('https://clerk.com/#/?test1=1&test2=2');
+  });
+
+  it('appends search params to the fragment by merging all passed in params', () => {
+    const base = 'https://clerk.com';
+    const url = buildURL(
+      { base, hashSearchParams: [new URLSearchParams({ test1: '1', test2: '2' }), { test3: '3' }] },
```

Comment:
What happens if you do this?

```ts
{ base, hashSearchParams: [new URLSearchParams({ test1: '1', test2: '2' }), { test2: '3' }] },
```

Reply:
The 2nd test2 will win, let me add a test just in case 👍🏻